core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Authenticator.java

```diff
@@ -36,23 +41,33 @@
 {
     private final OAuth2Service service;
     private final String principalField;
+    private final Optional<String> groupsField;
+    private final UserMapping userMapping;
 
     @Inject
     public OAuth2Authenticator(OAuth2Service service, OAuth2Config config)
     {
-        super(createUserMapping(config.getUserMappingPattern(), config.getUserMappingFile()));
         this.service = requireNonNull(service, "service is null");
         this.principalField = config.getPrincipalField();
+        groupsField = requireNonNull(config.getGroupsField(), "groupsField is null");
```

Comment:
nit: we can skip this rnn here and in other places as groups field is an empty `Optional` by default in the config.